ambari-common/src/main/python/ambari_commons/repo_manager/apt_parser.py

```diff
@@ -137,7 +137,7 @@ def packages_installed_reader(stream):
 
       line = line[2:].lstrip()
       data = line.partition(" ")
-      pkg_name = data[0]
+      pkg_name = data[0].partition(":")[0]  # for system packages in format "libuuid1:amd64"
```

Comment:
Any other sanity checks or logging here?  Assuming `[0]` exists in both these feels risky and relies on command output never changing.  That's gotten us in trouble with `rpm`/`yum` in the past.

Reply:
@ncole this is not `split`, i have used `partition` which not require the split character to exists and returns always a tuple of (left, separator, right).  i.e. if input string would be "teststring", it would return ('teststring', '', '')